src/main/java/appeng/client/render/model/MemoryCardBakedModel.java

```diff
@@ -0,0 +1,232 @@
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.IBakedModel;
+import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
+import net.minecraft.client.renderer.block.model.ItemOverrideList;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.world.World;
+import net.minecraftforge.common.model.TRSRTransformation;
+
+import appeng.api.implementations.items.IMemoryCard;
+import appeng.api.util.AEColor;
+import appeng.client.render.cablebus.CubeBuilder;
+import appeng.core.AELog;
+
+
+class MemoryCardBakedModel implements IBakedModel
+{
+	private static final AEColor[] DEFAULT_COLOR_CODE = new AEColor[] {
+			AEColor.TRANSPARENT, AEColor.TRANSPARENT, AEColor.TRANSPARENT, AEColor.TRANSPARENT,
+			AEColor.TRANSPARENT, AEColor.TRANSPARENT, AEColor.TRANSPARENT, AEColor.TRANSPARENT,
+	};
+
+	private final VertexFormat format;
+
+	private final IBakedModel baseModel;
+
+	private final TextureAtlasSprite texture;
+
+	private final AEColor[] colorCode;
+
+	private final Cache<CacheKey, MemoryCardBakedModel> modelCache;
+
+	private final ImmutableList<BakedQuad> generalQuads;
+
+	MemoryCardBakedModel( VertexFormat format, IBakedModel baseModel, TextureAtlasSprite texture )
+	{
+		this( format, baseModel, texture, DEFAULT_COLOR_CODE, createCache() );
+	}
+
+	private MemoryCardBakedModel( VertexFormat format, IBakedModel baseModel, TextureAtlasSprite texture, AEColor[] hash, Cache<CacheKey, MemoryCardBakedModel> modelCache )
+	{
+		this.format = format;
+		this.baseModel = baseModel;
+		this.texture = texture;
+		this.colorCode = hash;
+		this.generalQuads = ImmutableList.copyOf( this.buildGeneralQuads() );
+		this.modelCache = modelCache;
+	}
+
+	private static Cache<CacheKey, MemoryCardBakedModel> createCache()
+	{
+		return CacheBuilder.newBuilder()
+				.maximumSize( 100 )
+				.build();
+	}
+
+	@Override
+	public List<BakedQuad> getQuads( @Nullable IBlockState state, @Nullable EnumFacing side, long rand )
+	{
+
+		List<BakedQuad> quads = this.baseModel.getQuads( state, side, rand );
+
+		if( side != null )
+		{
+			return quads;
+		}
+
+		List<BakedQuad> result = new ArrayList<>( quads.size() + this.generalQuads.size() );
+		result.addAll( quads );
+		result.addAll( this.generalQuads );
+		return result;
+	}
+
+	private List<BakedQuad> buildGeneralQuads()
+	{
+		CubeBuilder builder = new CubeBuilder( this.format );
+
+		builder.setTexture( this.texture );
+
+		for( int x = 0; x < 4; x++ )
+		{
+			for( int y = 0; y < 2; y++ )
+			{
+				final AEColor color = this.colorCode[x + y * 4];
+
+				builder.setColorRGB( color.mediumVariant );
+				builder.addCube( 7 + x, 8 + ( 1 - y ), 7.5f, 7 + x + 1, 8 + ( 1 - y ) + 1, 8.5f );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this addition operation to a "float". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2184)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this addition operation to a "float". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2184)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this addition operation to a "float". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2184)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this addition operation to a "float". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2184)